### PR TITLE
Add partial like in item with variants

### LINKED ISSUE #230: Urgently need fix
When clicking on the star to save item, especially for furniture such as houseware and miscellaneous. When you click the star on other variant of the same item it makes the first variant of the item being starred together and if I were to go and select another to star under the same item but different variant the star of the first variant goes off and it goes on and off as you click on different variant of the same item and this is really such a hassle to work with. In the previous versions they work well so I’m not sure why is it happening this way now. 

Next issue is I’m not sure is it because of upgrading the app to a new version that causes this or when the app crashed or refreshed.l, the starred variants of the item that is not the first are all clean up meaning they are no longer starred and the record of them are gone which I find it frustrating because it take so much time to record/starred the stuff and if this were to happen every so often then it isn’t really helping at all. 

1 suggestion I thought of aside from all this issue that’s happening. Is it possible to make the main star if the item with different variant to only get starred when all variant of the collection is starred or vise versa when we click on the main star can it turn all the item in the collection to be starred? I think the point here is to separate them to their individual star but have a different star which is the main star to act on it’s on. Hope what I’m saying here y’all understand. 

Anyway last note I just wanted to say thanks guys for all the hard work y’all have put in to building  this app from the ground up so I really do appreciate it but at the same time I’m giving all this feedback is just to help y’all make things better for the community and the app! Once again thank you guys and keep up the good work!

### PR DESCRIPTION
**Pull request by renaudjenny**
* Add an half star to show that the item has been partially completed, meaning at least one of its variants is liked, but not all of them.
* Display bottom sheet variations if the item has variants instead of adding it to the collection without their variants.
* Fix #230

### CODE REVIEW

**Comment by renaudjenny** [comment]
LOL I wanted to show a video recording of the result, but it's now impossible to upload video on GitHub :(

**Comment by renaudjenny** [comment]
Screen Recording 2020-12-28 at 15 07 16

File: ACHNBrowserUI/Packages/Backend/Sources/Backend/Environments/UserCollection.swift [hunk 1]
```diff
@@ -132,29 +132,33 @@ public class UserCollection: ObservableObject {
...
             }
             variants[filename]?.append(variant)
             save()
-            return true
+            isAdded = true
         }
+
+        if !items.contains(item) && variants.completionStatus(for: item) != .unstarted {
+            // We add the item to the liked items if at least one of its variant is added
+            toggleItem(item: item)
```
Comment (Dimillian) [inline ACHNBrowserUI/Packages/Backend/Sources/Backend/Environments/UserCollection.swift:155]: You should `_ =` to silence the warning.

File: ACHNBrowserUI/Packages/Backend/Sources/Backend/Environments/UserCollection.swift [hunk 2]
```diff
@@ -132,29 +132,33 @@ public class UserCollection: ObservableObject {
...
+        } else if items.contains(item) && variants.completionStatus(for: item) == .unstarted {
+            // We remove the item from the liked items if there is none of its variants liked
+            toggleItem(item: item)
```
Comment (Dimillian) [inline ACHNBrowserUI/Packages/Backend/Sources/Backend/Environments/UserCollection.swift:158]: Same

**Comment by Dimillian** [comment]
I love the functionality and it works very good from the list, but it missing two things. 

* An animation when the bottom sheet is presented / dismissed. 
* When you're on the item detail screen, the favorite button in the navigation bar is doing nothing, it should also present this bottom sheet. 

Appart from that, awesome!

**Comment by renaudjenny** [comment]
Hi @Dimillian, yes good idea, I will work on that today 🙂

File: ACHNBrowserUI/Packages/Backend/Sources/Backend/Environments/UserCollection.swift (hunk 1 above)
Comment (renaudjenny) [reply ACHNBrowserUI/Packages/Backend/Sources/Backend/Environments/UserCollection.swift]: Done

File: ACHNBrowserUI/Packages/Backend/Sources/Backend/Environments/UserCollection.swift (hunk 2 above)
Comment (renaudjenny) [reply ACHNBrowserUI/Packages/Backend/Sources/Backend/Environments/UserCollection.swift]: Done, I'm also wondering if at the end of the day `@discardableResult` could come handy for the function. As it's not used in a lot of places, I think that's fine for now.

**Comment by renaudjenny** [comment]
@Dimillian, animation is done, I really like it! Let me know if you do too 😁

**Comment by Dimillian** [comment]
Love it! LGTM!

**Pull request merged**